pkg/auth/cred.go

```diff
@@ -17,19 +17,42 @@ limitations under the License.
 package auth
 
 import (
+	"context"
 	"fmt"
+	"time"
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
 	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
 	"k8s.io/klog/v2"
+	"knative.dev/pkg/logging"
 )
 
+type TokenWrapper struct {
+	cred azcore.TokenCredential
+}
+
+func NewTokenWrapper(cred azcore.TokenCredential) *TokenWrapper {
+	return &TokenWrapper{
+		cred: cred,
+	}
+}
+
+func (w *TokenWrapper) GetToken(ctx context.Context, options policy.TokenRequestOptions) (azcore.AccessToken, error) {
+	token, err := w.cred.GetToken(ctx, options)
+	if err != nil {
+		return azcore.AccessToken{}, err
+	}
+	logging.FromContext(ctx).Info("refreshing MDAL Token")
```

Comment:
Not sure how frequently this is called, but logging as `Info` seems excessive? I would also suggest something like "adjusting token expiration", this does not actually refresh it, does it?

Reply:
agree we should remove this log line as it's a hot path for refreshing. We probably should only log on error and in trace level for non-error

Reply:
moved to debug